### PR TITLE
Fixing Java MySQL sample devfile to be able to be run on OSIO

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
This PR fixes the Java MySQL devfile memory quotas in order to be able to run the devfile on OSIO

### What issues does this PR fix or reference?
Issue: https://github.com/redhat-developer/che-functional-tests/issues/556

### CODE REVIEW

File: devfiles/java-mysql/devfile.yaml [hunk 1]
```diff
@@ -24,7 +25,13 @@ components:
           -Duser.home=/home/user"
       - name: MAVEN_OPTS
         value: $(JAVA_OPTS)
+<<<<<<< Updated upstream
```
Comment (skabashnyuk) [inline devfiles/java-mysql/devfile.yaml]: is this line expected to be here?

File: devfiles/java-mysql/devfile.yaml [hunk 2]
```diff
@@ -24,7 +25,13 @@ components:
...
     memoryLimit: 700Mi
+=======
```
Comment (skabashnyuk) [inline devfiles/java-mysql/devfile.yaml]: is this line expected to be here?

File: devfiles/java-mysql/devfile.yaml [hunk 3]
```diff
@@ -24,7 +25,13 @@ components:
...
+      - name: PS1
+        value: $(echo ${0})\\$
+    memoryLimit: 768Mi
+>>>>>>> Stashed changes
```
Comment (skabashnyuk) [inline devfiles/java-mysql/devfile.yaml]: is this line expected to be here?

File: devfiles/java-mysql/devfile.yaml (hunk 1 above)
Comment (ScrewTSW) [reply devfiles/java-mysql/devfile.yaml]: that's weird. looks as if VisualStudioCode didn't rewrite the file
I'll fix it

File: devfiles/java-mysql/devfile.yaml (hunk 2 above)
Comment (ScrewTSW) [reply devfiles/java-mysql/devfile.yaml]: fixed

File: devfiles/java-mysql/devfile.yaml (hunk 3 above)
Comment (ScrewTSW) [reply devfiles/java-mysql/devfile.yaml]: fixed

**Comment by skabashnyuk** [comment]
CC @amisevsk

File: devfiles/java-mysql/devfile.yaml [hunk 4]
```diff
@@ -24,7 +25,9 @@ components:
           -Duser.home=/home/user"
       - name: MAVEN_OPTS
         value: $(JAVA_OPTS)
-    memoryLimit: 700Mi
+      - name: PS1
+        value: $(echo ${0})\\$
```
Comment (amisevsk) [inline devfiles/java-mysql/devfile.yaml]: `PS1` should not be included, it is set in the entrypoint on the patched image.

Comment (amisevsk) [reply devfiles/java-mysql/devfile.yaml]: [ref](https://github.com/eclipse/che-devfile-registry/blob/92624598ba6cb07007bae1f0281262fa93e5a34b/arbitrary-users-patch/entrypoint.sh#L10)

**Comment by amisevsk** [comment]
Looking at my quota usage with this devfile, I see `2.77Gi` used out of `3Gi` available; is there a reason we're not using the remaining `~230Mi` quota?

File: devfiles/java-mysql/devfile.yaml (hunk 4 above)
Comment (ScrewTSW) [reply devfiles/java-mysql/devfile.yaml]: will remove :)

**Comment by ScrewTSW** [comment]
> Looking at my quota usage with this devfile, I see `2.77Gi` used out of `3Gi` available; is there a reason we're not using the remaining `~230Mi` quota?

I can try to fiddle with it and submit a fix to the PR tomorrow.

**Comment by amisevsk** [comment]
Regardless I've tested and the lowered limit on java LS does not seem to interfere for the sample project. +1 to merge.

**Comment by amisevsk** [comment]
Looking at this more closely (output of `top` on the vscode-java container) I can see the language server being killed occasionally and restarted by the plugin. This is accompanied by logs like
```
[Info  - 2:59:18 PM] Connection to server got closed. Server will restart.
[Info  - 2:59:56 PM] Connection to server got closed. Server will restart.
[Error - 2:59:56 PM] Request textDocument/codeAction failed.
Error: Connection got disposed.
[...]
```

I also at one point ended up with two java language servers running simultaneously (though I don't know if that's an issue in `master`)

**Comment by ScrewTSW** [comment]
so upon investigation there are two constants consuming quota limits:
* JWT Proxy consumes 150MiB
* Machine exec pod consumes 150MiB
* Theia consumes **512MB** = 488.281MiB

and the plugins defined in the devfile:
* mysql : 256MiB (+40MiB new value **296MiB**)
* Java LSP : 1024MiB
* tools (maven) : 768MiB (+192MiB new value **960MiB**)

That leaves me 235.71875MiB to redistribute. After redistribution this leaves me exactly 3899392 bytes to spare or 3.71875MiB .

**Pull request merged**